docs/proposals/17-shoot-control-plane-migration-bad-case.md

```diff
@@ -0,0 +1,102 @@
+
+10. The special "copy" version of the `etcd-main` Etcd resource is deleted from the source seed, and as a result the `etcd-copy` Job is also deleted by `etcd-druid`.
+
+11. The additional "source" backup entry referencing the source seed backup container is deleted from the Garden cluster and the destination seed. As a result, its corresponding `source-etcd-backup` secret is also deleted from the destination seed.
+
+12. From this point, the reconciliation flow proceeds as already described in [GEP-7](07-shoot-control-plane-migration.md). This is safe, since the source seed cluster is no longer able to interfere with the shoot.
+
+## Handling Inability to Access the Backup Container
+
+The mechanism described above assumes that the `etcd-backup-restore` process in the source seed is able to access its backup container in order to take snapshots. If this is not the case, but an ownership change was detected, the `etcd-backup-restore` process still sets the readiness probe status of the main `etcd` container to 503, and kills the main `etcd` process as described above to ensure that any open connections from `kube-apiserver` are terminated. This effectively deactivates the source seed cluster to ensure that the ownership of the shoot can be passed to a different seed.
+
+Because of this, `etcd-backup-restore` process in the destination seed responsible for copying the snapshots can avoid waiting forever for a final full snapshot to appear. Instead, after a certain timeout has elapsed, it can proceed with the copying. In this situation, whatever latest snapshot is found in the source backup container will be restored in the destination seed. The shoot is still migrated to a healthy seed at the cost of losing the etcd data that accumulated between the point in time when the connection to the source backup container was lost, and the point in time when the source seed cluster was deactivated.
+
+When the connection to the backup container is restored in the source seed, a final full snapshot will be eventually taken. Depending on the stage of the restoration flow in the destination seed, this snapshot may be copied to the destination seed and restored, or it may simply be ignored since the snapshots have already been copied.
+
+## Migration Flow Adaptations
+
+Certain changes to the migration flow are needed in order to ensure that it is compatible with the [owner election](#owner-election--copying-snapshots) mechanism described above. Instead of taking a full snapshot of the source seed etcd, the flow deletes the owner DNS record by deleting the `owner` DNSRecord resource. This causes the ownership check by `etcd-backup-restore` to fail, and the final full snapshot to be eventually taken, so the migration flow waits for a final full snapshot to appear as the last step before deleting the shoot namespace in the source seed. This ensures that the reconciliation flow described above will find a final full snapshot waiting to be copied at step 6.
```

Comment:
The consequence is that the "good case" migration will become slower now, right?

Reply:
Yes, but only a little bit. The final full snapshot will appear when the source seed `etcd-backup-restore` process has detected the ownership change (the average time we observed was about 15 seconds, half of the 30 seconds polling interval) and the final full snapshot has been taken (very fast on small shoots, I assume not longer than a couple of minutes on really large shoots). So the slowdown in most cases should be be less than 1 minute.